src/test/java/org/springframework/data/redis/connection/jedis/StreamConvertersUnitTest.java

```diff
@@ -41,4 +51,201 @@ void shouldConvertIdle() {
 
 		assertThat(xPendingParams).hasFieldOrPropertyWithValue("idle", Duration.of(1, ChronoUnit.HOURS).toMillis());
 	}
+
+	@Nested // GH-3232
+	class ToXAddParamsShould {
+
+		@Test
+		void convertXAddOptionsWithMaxlen() {
+
+			RecordId recordId = RecordId.autoGenerate();
+			XAddOptions options = XAddOptions.trim(TrimOptions.maxLen(100));
+
+			XAddParams params = StreamConverters.toXAddParams(recordId, options);
+
+			assertThat(params).hasFieldOrPropertyWithValue("maxLen", 100L);
+		}
+
+		@Test
+		void convertXAddOptionsWithMinId() {
+
+			RecordId recordId = RecordId.autoGenerate();
+			XAddOptions options = XAddOptions.trim(TrimOptions.minId(RecordId.of("1234567890-0")));
+
+			XAddParams params = StreamConverters.toXAddParams(recordId, options);
+
+			assertThat(params).hasFieldOrPropertyWithValue("minId", "1234567890-0");
+		}
+
+		@Test
+		void convertXAddOptionsWithApproximateTrimming() {
+
+			RecordId recordId = RecordId.autoGenerate();
+			XAddOptions options = XAddOptions.trim(TrimOptions.maxLen(100).approximate());
+
+			XAddParams params = StreamConverters.toXAddParams(recordId, options);
+
+			assertThat(params).hasFieldOrPropertyWithValue("approximateTrimming", true);
+		}
+
+		@Test
+		void convertXAddOptionsWithExactTrimming() {
+
+			RecordId recordId = RecordId.autoGenerate();
+			XAddOptions options = XAddOptions.trim(TrimOptions.maxLen(100).exact());
+
+			XAddParams params = StreamConverters.toXAddParams(recordId, options);
+
+			assertThat(params).hasFieldOrPropertyWithValue("exactTrimming", true);
+		}
+
+		@Test
+		void convertXAddOptionsWithLimit() {
+
+			RecordId recordId = RecordId.autoGenerate();
+			XAddOptions options = XAddOptions.trim(TrimOptions.maxLen(100).approximate().limit(50));
+
+			XAddParams params = StreamConverters.toXAddParams(recordId, options);
+
+			assertThat(params).hasFieldOrPropertyWithValue("limit", 50L);
+		}
+
+		@Test
+		void convertXAddOptionsWithDeletionPolicy() {
+
+			RecordId recordId = RecordId.autoGenerate();
+			XAddOptions options = XAddOptions.trim(TrimOptions.maxLen(100).pendingReferences(StreamDeletionPolicy.keep()));
+
+			XAddParams params = StreamConverters.toXAddParams(recordId, options);
+
+			assertThat(params).hasFieldOrPropertyWithValue("trimMode",
+					redis.clients.jedis.args.StreamDeletionPolicy.KEEP_REFERENCES);
+		}
+
+		@Test
+		void convertXAddOptionsWithRecordId() {
+
+			RecordId recordId = RecordId.of("1234567890-0");
+			XAddOptions options = XAddOptions.none();
+
+			XAddParams params = StreamConverters.toXAddParams(recordId, options);
```

Comment:
A testing approach of testing that a component is not doing something quickly leads to a lot of complexity as the subsequent lines unfold into several assertions. Instead, I suggest testing for the expectation that the component is doing something we expect, namely, not adding any command arguments.

`XAddParams.addParams(CommandArguments)` would be a neat way to utilize public API for testing which command arguments are added.